### PR TITLE
Migrate to ESLint

### LINKED ISSUE #314: Migrate to ESLint
We need to replace tslint with eslint (see https://github.com/palantir/tslint/issues/4534).

### PR DESCRIPTION
**Pull request by timocov**
**Type of PR:** enhancement

**PR checklist:**

- [x] Addresses an existing issue: fixes #314
- [ ] Includes tests
- [ ] Documentation update

### CODE REVIEW

File: tests/e2e/serve-local-files.js [hunk 1]
```diff
@@ -75,7 +75,7 @@ function main() {
 }
 
 if (require.main === module) {
-	main().catch(e => {
+	main().catch((e) => {
```
Comment (Nipheris) [inline tests/e2e/serve-local-files.js:78]: Do we really need extra parenthesis for lambda parameters?

File: src/helpers/delegate.ts [hunk 2]
```diff
@@ -2,15 +2,15 @@ import { Callback, ISubscription } from './isubscription';
 
 interface Listener<T1, T2> {
 	callback: Callback<T1, T2>;
-	linkedObject?: object;
+	linkedObject?: unknown;
```
Comment (Nipheris) [inline src/helpers/delegate.ts:5]: Values of primitive types can be used as `linkedObject`?

Comment (timocov) [reply src/helpers/delegate.ts:5]: Yes, it could be any value actually.

File: tests/e2e/serve-local-files.js (hunk 1 above)
Comment (timocov) [reply tests/e2e/serve-local-files.js:78]: Not sure. Just wonder why I did even change it 😂 I can't find enabling `arrow-parens` rule though. I'll check.

Comment (timocov) [reply tests/e2e/serve-local-files.js:78]: That's strange. I'll enable `arrow-parens` rule with `as-needed` option.

Comment (timocov) [reply tests/e2e/serve-local-files.js]: Done in f041fff

**Pull request merged**